platform/ios/app/MBXEmbeddedMapViewController.m

```diff
@@ -0,0 +1,188 @@
+#import "MBXEmbeddedMapViewController.h"
+#import <Mapbox/Mapbox.h>
+
+typedef NS_ENUM(NSInteger, MBXEmbeddedControl) {
+    MBXEmbeddedControlZoom = 0,
+    MBXEmbeddedControlScroll,
+    MBXEmbeddedControlRotation,
+    MBXEmbeddedControlPitch
+};
+
+
+@interface MBXEmbeddedMapViewController () <UIScrollViewDelegate>
+
+@property (nonatomic, strong) MGLMapView *mapView;
+
+@end
+
+@implementation MBXEmbeddedMapViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    UIRotationGestureRecognizer *gestureRecognizer = [[UIRotationGestureRecognizer alloc] initWithTarget:self action:@selector(rotation:)];
+    
+    UIScrollView *scrollView = [[UIScrollView alloc] initWithFrame:self.view.frame];
+    scrollView.delegate = self;
+    [self.view addSubview:scrollView];
+    [scrollView addGestureRecognizer:gestureRecognizer];
+    
+    CGRect mapRect = CGRectMake(0, 0, CGRectGetWidth(scrollView.frame) * 2, CGRectGetHeight(scrollView.frame) * 2);
```

Comment:
It's true that https://github.com/mapbox/mapbox-gl-native/issues/7217 demonstrated the need for only vertical (and perhaps horizontal) gesture overrides. However, the solution implemented in this PR technically allows for all possible UIScrollView gestures to function instead of the built in map view gestures. This demo may not be realistic but it is comprehensive and I don't think it would make sense to throw away the pinch and rotate parts in iosapp.

That said, a more realistic example in mapbox/ios-sdk-examples would be helpful, I think.